scripts/travisci/run_unit_tests.py

```diff
@@ -0,0 +1,35 @@
+import os.path as osp
+import sys
+
+import nose2
```

Comment:
we should be able to do this purely with nose2

Reply:
I think the right way to do it is to run command nose2 tests directly so that every new pushes tests would be ran automatically. The only problem is how to exclude the tests we don’t want to run currently. The only way to select tests is by using test attributes while existing tests all don’t have attributes actually. I can also move unwanted tests out of the current directory but I think it would make project directory structure look dirty.

Reply:
The implementation of attributes doesn't seem hard to add (1 line in the test + 1 line in setup.cfg) and would be useful in the future (e.g. separating regression benchmarks from unit tests). I'd prefer this approach to maintaining another error-prone shell script (we already spend a lot of time debugging the existing ones).

Please correct me if it's harder than I imagine.

http://nose2.readthedocs.io/en/latest/plugins/attrib.html

Reply:
Okay I tried it--seems fine.

Steps:
1. Add the following to setup.cfg -- easy
    ```ini
    [unittest]
    plugins = nose2.plugins.attrib
              nose2.plugins.layers
    ```
1. Convert tests to actual Python unittest cases (necessary anyway) and then add an attribute -- easy
    test_networks.py - BEFORE
    ```python
    import lasagne.layers as L
    import numpy as np

    from garage.core import GRUNetwork
    from garage.misc import ext


    def test_gru_network():
        network = GRUNetwork(
            input_shape=(2, 3),
            output_dim=5,
            hidden_dim=4,
        )
        f_output = ext.compile_function(
            inputs=[network.input_layer.input_var],
            outputs=L.get_output(network.output_layer))
        assert f_output(np.zeros((6, 8, 2, 3))).shape == (6, 8, 5)
    ```
    test_networks.py -- AFTER
    ```python
    import unittest

    import lasagne.layers as L
    import numpy as np

    from garage.core import GRUNetwork
    from garage.misc import ext


    class TestNetworks(unittest.TestCase):
        def test_gru_network(self):
            network = GRUNetwork(
                input_shape=(2, 3),
                output_dim=5,
                hidden_dim=4,
            )
            f_output = ext.compile_function(
                inputs=[network.input_layer.input_var],
                outputs=L.get_output(network.output_layer))
            assert f_output(np.zeros((6, 8, 2, 3))).shape == (6, 8, 5)

        test_gru_network.broken = True
    ```
1. Run nose2 selecting an attribute (or its negative)
    ```sh
    (garage) rjulian@nyquist:~/code/garage/new_tests$ nose2 -c ../setup.cfg -A '!broken'
    .
    ----------------------------------------------------------------------
    Ran 1 test in 0.024s

    OK
    (garage) rjulian@nyquist:~/code/garage/new_tests$ nose2 -c ../setup.cfg
    E.
    ======================================================================
    ERROR: test_gru_network (test_networks.TestNetworks)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/rjulian/code/garage/new_tests/test_networks.py", line 15, in test_gru_network
        hidden_dim=4,
      File "/home/rjulian/code/garage/garage/core/network.py", line 263, in __init__
        b=l_output_flat.b,
      File "/home/rjulian/miniconda2/envs/garage/lib/python3.5/site-packages/lasagne/layers/dense.py", line 78, in __init__
        super(DenseLayer, self).__init__(incoming, **kwargs)
      File "/home/rjulian/miniconda2/envs/garage/lib/python3.5/site-packages/lasagne/layers/base.py", line 44, in __init__
        if any(d is not None and d <= 0 for d in self.input_shape):
      File "/home/rjulian/miniconda2/envs/garage/lib/python3.5/site-packages/lasagne/layers/base.py", line 44, in <genexpr>
        if any(d is not None and d <= 0 for d in self.input_shape):
    TypeError: unorderable types: tuple() <= int()

    ----------------------------------------------------------------------
    Ran 2 tests in 0.046s

    FAILED (errors=1)
    ```

NOTE: according to the nose2 documentation, it should find the config in setup.cfg, but for some reasons ours is not, so I pass it explicitly.